### PR TITLE
ssm-agent: 2.3.1319.0 -> 3.0.755.0

### PR DESCRIPTION
**Pull request by cole-h**
Co-authored-by: Cole Helbling <cole.e.helbling@outlook.com>

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114874` at 77879e39 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/912e287dd3bb85812bf3ae03923d73bd">ssm-agent</a></li></li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/networking/cluster/ssm-agent/default.nix:42:5:
```
   |
42 |     ./0001-Disable-NIC-tests-that-fail-in-the-Nix-sandbox.patch
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/networking/cluster/ssm-agent/default.nix:43:5:
```
   |
43 |     ./0002-version-gen-don-t-use-unnecessary-constants.patch
   |     ^
```
</li>
  </ul>
</details>

**Comment by cole-h** [comment]
(aarch64 build was fixed by dd3d0ca6c06710d1fb394ee8897a12f1e86aa2d8)

File: pkgs/applications/networking/cluster/ssm-agent/default.nix [hunk 1]
```diff
@@ -1,63 +1,109 @@
...
+    # Some tests use networking, so we skip them.
+    ./0001-Disable-NIC-tests-that-fail-in-the-Nix-sandbox.patch
+
+    # They used constants from another package that I couldn't figure
+    # out how to resolve, so hardcoded the constants.
+    ./0002-version-gen-don-t-use-unnecessary-constants.patch
+  ];
+
+  preConfigure = ''
+    rm -rf ./Tools/src/goreportcard
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/ssm-agent/default.nix:51]:
```suggestion
    rm -r ./Tools/src/goreportcard
```

**Review by SuperSandro2000 — DISMISSED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114874` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>ssm-agent</li>
</ul>
</details>

**Review by manveru — APPROVED** [review]
LGTM

**Comment by grahamc** [comment]
Pending some actual tests.

**Review by grahamc — APPROVED** [review]
SSM seems to work as well as it did before!

**Pull request merged**